ui/app/controllers/vault/cluster/access/oidc/clients/client/details.js

```diff
@@ -0,0 +1,23 @@
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+
+export default class OidcClientDetailsController extends Controller {
+  @service router;
+  @service flashMessages;
+
+  queryParams = ['tab'];
+  tab = 'details';
```

Comment:
Claire and I chatted about this on Zoom, but I'm curious what folks think about making the details route a tab. It looks like a tab in the UI, but it's technically a route.  You can find it in the router. And anytime I see the keywords param and tab, I'm looking for the url `?tab-name`. However, because the details route is a router, the URL is something like `oidc/clients/client/client-name/details` no query param necessary.  I understand the motivation for adding the query param here. The UI design makes it look like a tab, having it be a tab gives it some given benefits of being active when clicked and refreshing the model whenever the tab is changed. 

This is kind of a broader question moving forward as to our pattern for when should we use tabs? Is it okay to turn a route that works without one, into a tab for these benefits?

Reply:
We do already use the tab query param for much of our routing (global find for `tab:` and include `controllers`), so I don't think it's an antipattern. However, I think this is an issue of default naming -- when  we set the default here to `null` or an empty string and we link to it, the url shows up as `my-vault-url/oidc/client/details?tab=`, so I think this was just attempting to be explicit about what the default view gets you. 

My personal preference is that a new view is a different route (in the router) whether it's a tab or not, but that hasn't been our pattern so far. I think it could be worth broader discussion if we want to standardize, but we make heavy use of tabs elsewhere and we'd need to make the distinction very clear as to when we do or don't use them.